### ISSUE TITLE
gmsh 2.10.1 does not build with GUI on Mac OS 10.11 El Capitan

### ISSUE THREAD

**Issue by github4uw** [Homebrew/homebrew-science#2871]
Hi,

Why can gmsh 2.10.1 not be build with GUI enabled? Here is what I get:
$ brew install gmsh --without-opencascade --with-oce --with-fltk
==> Installing homebrew/science/gmsh with --without-opencascade, --with-oce, --with-fltk
fltk: OS X Yosemite or older is required.
Error: An unsatisfied requirement failed this build.

I am quite new to git/homebrew and ruby. So I would like to understand where the formula script
invokes this message? What is the "unsatisfied requirement"?

Is there a chance to get this work soon?

Best regards!

Mac OS 10.11 with Xcode 7.0.1

**Comment by dpo** [comment]
I'm not very knowledgeable about FLTK and it's hard to learn anything on OS support from their website, but the Homebrew formula seems to say that 10.11 isn't supported in the current version. It will be when FLTK 1.3.4 comes out.

**Comment by dpo** [comment]
Doesn't seem that I can do much about this. Feel free to reopen if there's more.

**Comment by github4uw** [comment]
I found this repository https://github.com/IngwiePhoenix/FLTK for fltk 1.3.4.
It explicitly mentions support for Mac OS 10.11. 
Could this one used as a dependency? Can I just install it in addition to the fltk 1.3.3 and then rebuild gmsh?

**Comment by dpo** [comment]
I just put this together: https://gist.github.com/5b906b02e585d6f43eec. It builds the development snapshot, which says it supports El Capitan. It builds for me, but I'm running 10.9, so please check.

See https://github.com/Homebrew/homebrew/pull/44806 for a HEAD build, but there seem to be troubles.

**Comment by github4uw** [comment]
Thanks for your effort. I built the fltk snapshot and it seems to work. 
$ brew info fltk
fltk: stable 1.3.3
Cross-platform C++ GUI toolkit
http://www.fltk.org/
/usr/local/Cellar/fltk/1.3.x-r10866 (322 files, 5.8M) *
  Built from source
From: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/fltk.rb

So fltk is active as 1.3.x-r10866 after using brew switch <version>. However, when I try to reinstall gmsh 2.10.1 I still get the error:
$ brew reinstall gmsh --without-opencascade --with-oce --with-fltk
==> Reinstalling homebrew/science/gmsh with --without-opencascade, --with-oce, --with-fltk
fltk: OS X Yosemite or older is required.
Error: An unsatisfied requirement failed this build.

**Comment by github4uw** [comment]
Ok here comes the strange thing. I did a "trick change" in the gmsh.rb formula:
About line 84:
84 # Build with or without GUI
85 args << "-DENABLE_FLTK=OFF" if build.without? "fltk"
I changed OFF to ON ... and then I did:
$ brew reinstall gmsh --without-opencascade --with-oce
==> Reinstalling homebrew/science/gmsh with --without-opencascade, --with-oce

So I built gmsh without the --with-fltk option which should give non-GUI gmsh. But I had DENALE_FLTK=ON which should negate the missing --with-fltk option. And to a newbies surprise invoking $ gmsh in the terminal starts gmsh WITH working GUI. So may be the --with-fltk option does something strange ...
